scripts/constants.sh

```diff
@@ -34,17 +34,40 @@ else
     SUBNET_EVM_COMMIT="$(git --git-dir="$SUBNET_EVM_PATH/.git" rev-parse HEAD || :)"
 fi
 
+# Don't export them as they're used in the context of other calls
+if [[ -z ${AVALANCHE_VERSION:-} ]]; then
+  # Get module details from go.mod
+  MODULE_DETAILS="$(go list -m "github.com/ava-labs/avalanchego" 2>/dev/null)"
+
+  # Extract the version part
+  AVALANCHE_VERSION="$(echo "${MODULE_DETAILS}" | awk '{print $2}')"
+
+  # Check if the version matches the pattern where the last part is the module hash
+  # v*YYYYMMDDHHMMSS-abcdef123456
+  #
+  # If not, the value is assumed to represent a tag
+  if [[ "${AVALANCHE_VERSION}" =~ ^v.*[0-9]{14}-[0-9a-f]{12}$ ]]; then
+    # Extract module hash from version
+    MODULE_HASH="$(echo "${AVALANCHE_VERSION}" | grep -Eo '[0-9a-f]{12}$')"
+
+    # The first 8 chars of the hash is used as the tag of avalanchego images
+    AVALANCHE_VERSION="${MODULE_HASH::8}"
+  fi
+fi
+
 # Shared between ./scripts/build_docker_image.sh and ./scripts/tests.build_docker_image.sh
-DOCKERHUB_TAG=${SUBNET_EVM_COMMIT::8}
+DOCKERHUB_TAG="${SUBNET_EVM_COMMIT::8}_${AVALANCHE_VERSION}"
+# WARNING: this will use the most recent commit even if there are un-committed changes present
+BUILD_IMAGE_ID=${BUILD_IMAGE_ID:-"${CURRENT_BRANCH}_${AVALANCHE_VERSION}"}
 
 echo "Using branch: ${CURRENT_BRANCH}"
 
 # Static compilation
 STATIC_LD_FLAGS=''
 if [ "${STATIC_COMPILATION:-}" = 1 ]; then
-    export CC=musl-gcc
-    command -v $CC || (echo $CC must be available for static compilation && exit 1)
-    STATIC_LD_FLAGS=' -extldflags "-static" -linkmode external '
+  export CC=musl-gcc
+  command -v $CC || (echo $CC must be available for static compilation && exit 1)
+  STATIC_LD_FLAGS=' -extldflags "-static" -linkmode external '
```

Comment:
Question for future reference, what shell script formatting tool are you using? We could also use that to have consistent shell formatting and not have out of scope diffs changing the formatting of scripts (like here)

Reply:
I frankly don't know why my editor keep changing these. I'm using VS code and dont have any special formatter (at least not for script files). This one looks like a tab vs spaces issue.